### PR TITLE
Make all tests discoverable by QtCreator

### PR DESCRIPTION
**Pull request by arBmind**
QtCreator parses the source files to discover tests.
When we arrange the tests the right way - they can be found.

It seems too many or too big macros are an issue.

### CODE REVIEW

File: tests/qt/qmetamethod/tst_qmetamethod.cpp [hunk 1]
```diff
@@ -39,16 +39,22 @@ class tst_QMetaMethod : public QObject
     W_OBJECT(tst_QMetaMethod)
 
 private slots:
-    void method_data(); W_SLOT(method_data, W_Access::Private)
-    void method(); W_SLOT(method, W_Access::Private)
+    void method_data();
```
Comment (ogoffart) [inline tests/qt/qmetamethod/tst_qmetamethod.cpp]: Why is that required?  
I think it is better to have the W_SLOT next to the method, this way we can easily see that the W_SLOT macro was not forgotten.

Comment (arBmind) [reply tests/qt/qmetamethod/tst_qmetamethod.cpp]: It's not absolutely required. If QtCreator finds one slot, it will run all tests.

But QtCreator will not be able to find the other slots that way. Therefore you cannot limit test runs to these tests.

Comment (ogoffart) [reply tests/qt/qmetamethod/tst_qmetamethod.cpp]: I am not using QtCreator, so i'm not sure what you mean by QtCreator "finding" the test.
Why should QtCreator find the test? Is it not just running the tests by compiling and running the test program?

Comment (arBmind) [reply tests/qt/qmetamethod/tst_qmetamethod.cpp]: > I am not using QtCreator

That's strange. There are tests that were structured like that. - I made it consistent.

> Is it not just running the tests by compiling and running the test program?

Most build systems support running tests, but QtCreator does not support this.
You can select one executable and run it or use the test runner.
The test runner only runs executables with tests, that QtCreator had discovered before. 

> Why should QtCreator find the test?

See https://doc.qt.io/qtcreator/creator-autotest.html
Unfortunately QtCreator does not ask the build system but scans the source files using the C++ Code Model.
See: https://github.com/qt-creator/qt-creator/blob/master/src/plugins/autotest/qtest/qttestparser.cpp
If a project contains at least one test - it is shown and run by the test runner.
If all tests are discovered it's possible to select the tests to run.

Maybe it's not the best idea to use Verdigris to write the tests. If something in Verdigris is wrong, it's possible that unit tests pass because not tests were run.

File: tests/qt/qmetaobject/tst_qmetaobject.cpp [hunk 2]
```diff
@@ -407,6 +374,39 @@ private slots:
 
     void notifySignalsInParentClass();
 
+public:
```
Comment (ogoffart) [inline tests/qt/qmetaobject/tst_qmetaobject.cpp]: Is this change also necessary?  I want to avoid diverging away from the upstream tests)  as it make merging more difficule.

**Comment by ogoffart** [comment]
I don't really like this change because it will cause problem to maintain these tests:
These test are somehow sync'ed form Qt upstream (see the `tests/qt/README` file) and so i'd avoid departing away from the upstream tests. 
And the reason why the W_SLOT are directly next to the slot declaration has the advantage to force a merge conflict so we do not forget to add the W_SLOT when a slot is added upstream. (otherwise the test would be silently ignored (as in: not run).

I'm Ok removing the comments in slot if that helps though.
